src/storage/src/render/sources.rs

```diff
@@ -371,12 +375,35 @@ where
                             let (previous, previous_token, feedback_handle) =
                                 if Timestamp::minimum() < upper_ts {
                                     let as_of = Antichain::from_elem(upper_ts.saturating_sub(1));
+                                    let StorageMaxInflightBytesConfig {
+                                        max_in_flight_bytes_default,
+                                        max_in_flight_bytes_cluster_size_percent,
+                                        cluster_size_memory_map,
+                                    } = &storage_state
+                                        .dataflow_parameters
+                                        .storage_dataflow_max_inflight_bytes_config;
+
+                                    let current_cluster_memory_limit = cluster_size
+                                        .map(|size| cluster_size_memory_map.get(size))
+                                        .flatten();
+
+                                    let max_inflight_bytes_for_current_cluster: Option<usize> =
+                                        if let (Some(current_cluster_memory), Some(percent)) = (
+                                            current_cluster_memory_limit,
+                                            max_in_flight_bytes_cluster_size_percent,
+                                        ) {
+                                            Some(current_cluster_memory * percent / 100)
+                                        } else {
+                                            None
+                                        };
+
+                                    let backpressure_max_inflight_bytes =
+                                        max_inflight_bytes_for_current_cluster
+                                            .or(*max_in_flight_bytes_default);
```

Comment:
can we factor this into a function? If we have time can we add some simple unit tests?

Reply:
done